### PR TITLE
[C++] Update README for linting

### PR DESCRIPTION
**Pull request by pitrou**
There are hidden gotchas when trying to run the linting Makefile targets, mention them.

See issue #1514 (I'm not sure it's required to open a JIRA entry for such a trivial issue?).

### CODE REVIEW

**Comment by pitrou** [comment]
Note we could also update `cpplint` to the newer py3-compatible forked version from https://github.com/cpplint/cpplint.

File: cpp/README.md [hunk 1]
```diff
@@ -250,6 +250,23 @@ Logging IWYU to /tmp/arrow-cpp-iwyu.gT7XXV
...
+You can check your code abides by that coding style by running:
+
+    make lint
+
+You can also fix any formatting errors automatically:
+
+    make format
+
+These commands require `clang-format-4.0` (and not any other version).
+You may found the required packages for Debian/Ubuntu on https://apt.llvm.org/.
```
Comment (wesm) [inline cpp/README.md]: Thanks for adding this! Maybe also add http://releases.llvm.org/download.html which has binaries for Windows, macOS, and other Linuxes

**Review by wesm — APPROVED** [review]

File: cpp/README.md (hunk 1 above)
Comment (pitrou) [reply cpp/README.md]: Done.

**Review by wesm — APPROVED** [review]
+1. Opened https://issues.apache.org/jira/browse/ARROW-2035 about updating the vendored cpplint

**Pull request merged**